### ISSUE TITLE
How do I currently disable and hide fields in the checkout?

### ISSUE THREAD

**Issue by KPABOOKING** [woothemes/woocommerce#54069]
### Discussed in https://github.com/woocommerce/woocommerce/discussions/54063

<div type='discussions-op-text'>

<sup>Originally posted by **KPABOOKING** December 27, 2024</sup>
I use the checkout block and have tried many codes, until recently everything worked, after the update - making optional and hiding fields is just not possible.
I tried this code as well:

```
add_filter('woocommerce_get_country_locale', function( $locale ) {
	foreach ( $locale as $key => $value ) {
		$locale[ $key ]['address_1'] = [
			'required' => false,
			'hidden'   => true,
		];

		$locale[ $key ]['postcode'] = [
			'required' => false,
			'hidden'   => true,
		];

		$locale[ $key ]['city'] = [
			'required' => false,
			'hidden'   => true,
		];

		$locale[ $key ]['state'] = [
			'required' => false,
			'hidden'   => true,
		];
	}

	return $locale;
});

```

This code is presented to us as official - but it doesn't work.
https://developer.woocommerce.com/docs/cart-and-checkout-removing-checkout-fields/
I only sell to one country (Russia).
It didn't help</div>

**Comment by wavvves** [comment]
Hi @KPABOOKING thank you for reporting this.

I tested that snippet on the latest WC, and it worked, removing the indicated fields. Can you please provide more info on how and where are you adding the filter?

Can you also test the code using the [Code Snippets](https://wordpress.org/plugins/code-snippets/) extension? Does it work for you that way?
In that case we may need to adjust where the code is being placed, as it might not run in time to apply the changes.

Also, please describe in more detail the actual use case you have for disabling and hiding certain so I can try and help you achieve the intended purpose. (You mentioned the code you had functioned before updating WooCommerce, please provide details and steps you took to add it)

**Comment by KPABOOKING** [comment]
I ran the code through functions.php
I'll have to try Code Snippets

**Comment by webprom** [comment]
I'm using the code above, and it works for most countries, but it's not working for some, like Armenia, Russia, etc. Very weird...

**Comment by KPABOOKING** [comment]
@webprom Yes, it doesn't work for Russia, maybe we don't choose the country code the right way?

**Comment by webprom** [comment]
@KPABOOKING don't know... many other countries didn't work too. I installed "[Checkout Field Editor for WooCommerce](https://wordpress.org/plugins/woo-checkout-field-editor-pro/)" plugin and it worked for all countries. So you may examine their code, or just use it :)

**Comment by KPABOOKING** [comment]
@webprom As far as I remember the plugin doesn't work with Woocommerce Checkout Block, only with standard checkout [woocommerce_checkout], or Checkout Manager started to support Woo Checkout Block (Widget?).

**Comment by webprom** [comment]
@KPABOOKING yes, they support it now.
